pkg/component/component.go

```diff
@@ -15,18 +15,22 @@ limitations under the License.
 */
 package component
 
-import "github.com/k0sproject/k0s/pkg/apis/k0s.k0sproject.io/v1beta1"
+import (
+	"context"
+
+	"github.com/k0sproject/k0s/pkg/apis/k0s.k0sproject.io/v1beta1"
+)
 
 // Component defines the interface each managed component implements
 type Component interface {
 	Init() error
-	Run() error
+	Run(context.Context) error
```

Comment:
@jnummelin  @ncopa  Just thinking aloud outside of the PR context, I wonder do we want to have the same definition in Stop function to introduce context based timeouts for the graceful shutdown?

Reply:
hmm, not sure. We're now going into a direction where we have a "global" context anyways and thus it feels like the general `Stop()` function might not even be needed. I mean that when we `Run()` as component with a ctx the component needs to get hold of the given context. Then the component can stop all-the-things based on `<-ctx.Done()`, right? The "global" context is the one we use to signal EVERYTHING to stop, right?

But I agree, these parts go bit outside of this PR

Reply:
yeah. lets do that outside this PR